lib/upipe-freetype/upipe_freetype.c

```diff
@@ -489,6 +580,117 @@ static void upipe_freetype_input(struct upipe *upipe, struct uref *uref,
     }
 }
 
+/** @internal @This parses an RGBA color.
+ *
+ * @param upipe description structure of the pipe
+ * @param value RGBA value to parse
+ * @param rgba array filled with the r, g, b and a value
+ * @return an error code
+ */
+static int upipe_freetype_parse_rgba(struct upipe *upipe,
+                                     const char *value,
+                                     uint8_t def,
+                                     uint8_t rgba[4])
+{
+    memset(rgba, def, 4);
+
+    if (!value)
+        return UBASE_ERR_NONE;
+
+    int ret = sscanf(value, "rgb(%hhu, %hhu, %hhu)",
+                 &rgba[0], &rgba[1], &rgba[2]);
+    if (ret == 3) {
+        rgba[3] = 0xff;
+        return UBASE_ERR_NONE;
+    }
+
+    float alpha;
+    ret = sscanf(value, "rgba(%hhu, %hhu, %hhu, %f)",
+                 &rgba[0], &rgba[1], &rgba[2], &alpha);
+    if (ret == 4) {
+        rgba[3] = 0xff * alpha;
+        return UBASE_ERR_NONE;
+    }
+
+    return UBASE_ERR_INVALID;
+}
+
+/** @internal @This converts rgba color to yuva.
+ *
+ * @param upipe description structure of the pipe
+ * @param rgba rgb and alpha color to convert
+ * @param yuva yuv and alpha color converted
+ * @return an error code
+ */
+static int upipe_freetype_rgba_to_yuva(struct upipe *upipe,
```

Comment:
If this not specific to freetype, it should maybe be in ubuf_pic.c.

Reply:
Fixed